service/history/decisionTaskHandler.go

```diff
@@ -307,10 +328,26 @@ func (handler *decisionTaskHandlerImpl) handleDecisionCompleteWorkflow(
 	)
 
 	if handler.hasUnhandledEventsBeforeDecisions {
-		handler.failDecision = true
-		handler.failDecisionCause = workflow.DecisionTaskFailedCauseUnhandledDecision.Ptr()
+		return handler.handlerFailDecision(workflow.DecisionTaskFailedCauseUnhandledDecision, "")
+	}
+
+	err := handler.attrValidator.validateCompleteWorkflowExecutionAttributes(attr)
+	if err != nil {
+		if _, ok := err.(*workflow.BadRequestError); ok {
```

Comment:
can we make this a method, `processBadRequestError` or something, that either modifies the error if its bad request or returns it as is?

Reply:
I guess it has to take in the fail decision reason..

Reply:
already done